### PR TITLE
New branch (+ changes) for Raspberry Pi

### PR DESCRIPTION
**Pull request by enoch85**
As discussed via email.

So the only thing that seems to fail is smbclient, which I find odd as it's already installed via PECL.

**For testing purposes, this is the setup:**

- Raspberry Pi 4 (4 GB RAM or more)
- Ubuntu 20.04 Server image from here: https://ubuntu.com/download/raspberry-pi (64-bit)
- Run the install script as usual

I don't have a RPi available to me right now, so anyone that have one laying around can test this. Would be awesome!

I general it actually seems like the scripts are directly applicable on RPi 4, which is super cool!

Thanks to @BigDogJonx for giving this a try from the get go.

### CODE REVIEW

**Comment by BigDogJonx** [comment]
As far as smbclient, I believe the issue also resides on your normal VM image as well (its not just only related to this RPi).  To test this I just downloaded the 40gb OVA, then installed it on VMware workstation.  Then I basically just added a single SMB share from my Windows share using External Storage app.  The share itself works in that it appears, you can access the share/folders/files, however as mentioned previously via email, the issue is that it doesnt index any of the files, the share/folders will all show "pending" in size.  

When you run a occ files:scan --all, it doesnt index the files which can be seen here in this screenshot.  The 5folders/16files are the default ones inside Nextcloud.  There should be well into the thousands with that share I added.
https://imgur.com/QzKH0s4

I believe this needs php-smbclient as it was there previous scripts I believe and allowed to be installed on Ubuntu 18.  Now with Ubuntu 20 you cannot install that package using apt-get.  

So only workaround I found was to run the following cmds based on this link from Owncloud.

https://doc.owncloud.com/server/admin_manual/enterprise/external_storage/windows-network-drive_configuration.html

Install it based on that site which is basically the following:
```
git clone git://github.com/eduardok/libsmbclient-php.git
cd libsmbclient-php ; phpize
./configure
make
sudo make install
```

Create a file with the following in there: (sudo pico /etc/php/7.4/mods-available/smbclient.ini)
`extension="smbclient.so"`

Then enable it:
`sudo phpenmod -v ALL smbclient`

Once you do all this, then if I go back and now run the same occ files:scan -all cmd, here is what I get:
https://imgur.com/ovKhRmb

So without that smbclient, it doesnt allow it to fully index the files, once this is installed now all folders show their actual sizes in nextcloud versus showing pending.  Again this happens on your normal VMware OVA image, so I had to do the same thing on RPi4 with Ubuntu 20.04 (RPi version).  Then that portion worked.  So I think this smbclient is an issue on both sides.

-----------------------------------------

And finally the LetEncrypt portion of your script will fail on the RPi as certbot isnt supported on arm.  So the workaround there was to run these cmds to get it to work:
Will have to run this manually as it doesnt work with the TechandMe script:
https://certbot.eff.org/lets-encrypt/ubuntufocal-apache
```
sudo apt-get update
sudo apt-get install software-properties-common
sudo add-apt-repository universe
sudo apt-get update
sudo apt-get install certbot python3-certbot-apache
```

**Comment by BigDogJonx** [comment]
One last piece I noticed was php-imagick.  I was able to resolve it by just installing it via sudo apt-get install php-imagick.  This issue also resides on the normal VM image too.  However just noticed that cmd isnt working now, saying package has been obsoleted.  Pretty sure I installed this.

Also since I have my Nextcloud18 VM image still running, I checked that one and both php-smbclient and php-imagick were installed (using dpkg -l).  So thats probably why we didnt see that issue on there on Ubuntu 18.

**Comment by enoch85** [comment]
@BigDogJonx 

>One last piece I noticed was php-imagick

There a reason for that: https://github.com/nextcloud/server/issues/13099

> And finally the LetEncrypt portion of your script will fail on the RPi as certbot isnt supported on arm

It's installed via snap: https://github.com/nextcloud/vm/blob/master/lib.sh#L502-L516

Also, I have a RPi 4 in use for another thing here at home. Just tested and it works: 

```
root@raspberrypi:~# snap install --beta certbot --classic
certbot (beta) 1.6.0 from Certbot Project (certbot-eff✓) installed
```

> Ubuntu 20 you cannot install that package using apt-get.

I'd rather not install any git repo stuff manually because I want to keep it clean. It will come to Ubuntu 20.04 eventually, and when it does I will add it. As you already mentioned, the share works, but it doesn't index. The most important thing here is that it works.

File: README.md [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-============
+Nextcloud RPi
+=============
 
 Server installation. Simplified. :cloud:
 --------------------------------
 
 #### Interactive Guidance
-> **The Nextcloud VM** — _(aka **N**ext**c**loud **V**irtual **M**achine_ or _**NcVM**)_ — helps you create a personal or corporate [Nextcloud Server] _faster_ and _easier._ Fundamentally, NcVM is a carefully crafted _family_ of [\*nix] scripts which interactively guide you through a quality-controlled installation to obtain an [A+ security-rated] Nextcloud instance.
+> **The Nextcloud RPi** — _(aka **N**ext**c**loud **V**irtual **M**achine_ or _**NcRPi**)_ — helps you create a personal or corporate [Nextcloud Server] _faster_ and _easier._ Fundamentally, NcRpi is a carefully crafted _family_ of [\*nix] scripts which interactively guide you through a quality-controlled installation to obtain an [A+ security-rated] Nextcloud instance.
```
Comment (enoch85) [inline README.md:8]: @BigDogJonx Please be consistent and use NcRPi all over.

File: nextcloud_install_production.sh [hunk 2]
```diff
@@ -263,7 +263,7 @@ check_command apt install -y \
     php"$PHPVER"-bz2 \
     php"$PHPVER"-bcmath \
     php-pear
-    # php"$PHPVER"-imagick \
+    php"$PHPVER"-imagick \
```
Comment (enoch85) [inline nextcloud_install_production.sh:266]: Please revert this, we don't want to install imagick.

**Comment by enoch85** [comment]
> and seems like everything worked on the TLS side.

Great!

Can you please test to install all the apps as well? Just run `sudo bash /var/scripts/menu.sh` and run every option there is.

Please report back if you find anything that isn't working.

THANKS! :)

**Comment by BigDogJonx** [comment]
> Can you please test to install all the apps as well? Just run `sudo bash /var/scripts/menu.sh` and run every option there is.
> 
> Please report back if you find anything that isn't working.
> 
> THANKS! :)

Ran the script, checked them all, took about 34mins to run through all of it, ended up doing a screen recording capture off my macbook to then review since stuff flies by fast, and only one I saw fail was BitWarden.  Here is a screenshot I captured to show its failure, but every other piece installed fine and gave me the screen showing it was successfully installed.

https://i.imgur.com/TtgeXrA.png

**Comment by enoch85** [comment]
Super cool! 

So one last thing, please review all the lines of code in this new branch (or simply grep for "VM" so see it there's anything we would have to replace to RPi instead.

Thanks!

**Comment by enoch85** [comment]
@BigDogJonx 

Hmm, just looked at that repo and it clearly says: https://github.com/eduardok/libsmbclient-php#installation-from-pecl

So, the method w are using already should work.

Could you please investigate if we manually would need to add something like: `extension="smbclient.so"` in `/etc/php/7.4/....` like you were writing here:

> Create a file with the following in there: (sudo pico /etc/php/7.4/mods-available/smbclient.ini)
extension="smbclient.so"

> Then enable it:
sudo phpenmod -v ALL smbclient

**Comment by enoch85** [comment]
After some more investigation I can see that it should already be enabled: https://github.com/nextcloud/vm/blob/master/nextcloud_install_production.sh#L478-L486

This is really odd...

**Comment by BigDogJonx** [comment]
> After some more investigation I can see that it should already be enabled: https://github.com/nextcloud/vm/blob/master/nextcloud_install_production.sh#L478-L486
> 
> This is really odd...

Alright now Im getting somewhere on here.  And I found the issue I believe.  

First yes the above section shows it should be enabled, but thats in the php.ini.  

Now if I go and run the cmd of `sudo phpenmod -v ALL smbclient` it tells me that the smbclient.ini is missing from the /etc/php/7.4/mods-available directory.  So that led me to believe its not using that extension at all.

So to fix it all I did was add that extension to smbclient.ini and enable it.  I simple added the `extension="smbclient.so"` in the smbclient.ini file (which I had to create one since one didnt exist) at /etc/php/7.4/mods-available, then ran the `sudo phpenmod -v ALL smbclient` and then it worked.  `sudo -u www-data /var/www/nextcloud/occ files:scan --all` now works as it should, all files get indexed, as mentioned before if I run the occ file:scan cmd without doing any of these changes after jsut simply adding the SMB share in Nextcloud using External storage, it will not index any files off the SMB share, it just acts as if it doesnt exist since it keeps showing the same 5 directories and 16 files only indexed.

**Comment by enoch85** [comment]
@BigDogJonx Please try the latest master, I just added your fix.

Also, please pull the latest master to your repo.

Thanks!

**Comment by enoch85** [comment]
Hmm, the files should not conflict... Please do a new pull since I made more changes.

Thanks!

**Comment by enoch85** [comment]
It should be something like: `git fetch upstream`

https://reflectoring.io/github-fork-and-pull/#updating-your-fork

**Comment by BigDogJonx** [comment]
> It should be something like: `git fetch upstream`
> 
> https://reflectoring.io/github-fork-and-pull/#updating-your-fork

Dammit, lol.  Let me see what I messed up on.  This GitHub Desktop (windows) seems to suck in my opinion unless I have no idea what Im doing (which easily could be the case since I clearly am new at this).

I know the original fork I did on this on my BigDogJonx Github was back in end of May I believe.  Been racking my brain on why this damn github desktop or even the web version refuses to update the fork to your latest.

Let me read that link and see what Im doing wrong, worst case I guess I can redo all my changes.  I can only blame myself on this since I have zero experience on the code management front lol.

**Comment by enoch85** [comment]
@BigDogJonx I just cloned your repo and followed the instructions, it works. 

So please use Linux command line and clone your fork, and follow the instructions. :)

**Comment by enoch85** [comment]
Hey, I just made it easier for you and pushed the changes from master into RPi.

Let's close this PR and make a new one if needed. Please base your next PR on the `rpi` branch.

Thanks!

**Pull request closed without merging**